array_api_tests/test_elementwise_functions.py

```diff
@@ -851,57 +847,65 @@ def test_round(x):
     # This is the same strategy used in the mask in the
     # test_round_special_cases_one_arg_two_integers_equally_close special
     # cases test.
-    floor = _array_module.floor(x)
-    ceil = _array_module.ceil(x)
-    over = _array_module.subtract(x, floor)
-    under = _array_module.subtract(ceil, x)
-    round_down = less(over, under)
-    round_up = less(under, over)
+    floor = xp.floor(x)
+    ceil = xp.ceil(x)
+    over = xp.subtract(x, floor)
+    under = xp.subtract(ceil, x)
+    round_down = xp.less(over, under)
+    round_up = xp.less(under, over)
     assert_exactly_equal(a[round_down], floor[round_down])
     assert_exactly_equal(a[round_up], ceil[round_up])
 
 @given(numeric_scalars)
 def test_sign(x):
-    # a = _array_module.sign(x)
+    # a = xp.sign(x)
     pass
 
 @given(floating_scalars)
 def test_sin(x):
-    # a = _array_module.sin(x)
+    # a = xp.sin(x)
     pass
 
 @given(floating_scalars)
 def test_sinh(x):
-    # a = _array_module.sinh(x)
+    # a = xp.sinh(x)
     pass
 
 @given(numeric_scalars)
 def test_square(x):
-    # a = _array_module.square(x)
+    # a = xp.square(x)
     pass
 
 @given(floating_scalars)
 def test_sqrt(x):
-    # a = _array_module.sqrt(x)
+    # a = xp.sqrt(x)
     pass
 
 @given(two_numeric_dtypes.flatmap(lambda i: two_array_scalars(*i)))
 def test_subtract(args):
     x1, x2 = args
     sanity_check(x1, x2)
-    # a = _array_module.subtract(x1, x2)
+    # a = xp.subtract(x1, x2)
 
 @given(floating_scalars)
 def test_tan(x):
-    # a = _array_module.tan(x)
+    # a = xp.tan(x)
     pass
 
 @given(floating_scalars)
 def test_tanh(x):
-    # a = _array_module.tanh(x)
+    # a = xp.tanh(x)
     pass
 
-@given(numeric_scalars)
+@given(xps.arrays(dtype=numeric_dtypes, shape=xps.array_shapes()))
 def test_trunc(x):
-    # a = _array_module.trunc(x)
-    pass
+    out = xp.trunc(x)
+    assert out.dtype == x.dtype, f"{x.dtype=!s} but {out.dtype=!s}"
+    assert out.shape == x.shape, f"{x.shape} but {out.shape}"
+    if x.dtype in integer_dtype_objects:
+        assert xp.all(xp.equal(x, out)), f"{x=!s} but {out=!s}"
+    else:
+        finite_mask = xp.isfinite(out)
+        for idx in ndindex(out.shape):
```

Comment:
You should be able to write this test without using ndindex, using the same kind of thing as test_floor and test_ceil.

Reply:
Very useful, thanks.